app/src/components/card/searchCard.tsx

```diff
@@ -0,0 +1,138 @@
+"use client";
+import React from "react";
+import {
+  Card,
+  CardHeading,
+  CardContent,
+  StatusBadge,
+  TagSet,
+  Flex,
+  Text,
+  Box,
+} from "@nypl/design-system-react-components";
+import SearchCardType, {
+  SearchResultRecordType,
+} from "@/src/types/SearchCardType";
+
+export interface SearchCardProps {
+  result: SearchCardType;
+  keywords: string;
+}
+
+const onSiteMaterialBadge = (recordType: SearchResultRecordType) => {
```

Comment:
As you keep working on this, would it help to have small classes for each of the record types? I'm mostly wondering if it will be helpful to have a class for an item, for example, so that you don't have to check `recordType === "Item"` or similar for the `contentType === "Image"` below. 

Or, how about a new class property for the badge display text in order to move the logic to the class. It would be very similar to `this.url` where it's also checking `data.recordType === "item"`. As much as possible to move logic to the class and "simply" display content in the component can be helpful for readability.